### ISSUE TITLE
Snowflake Key Pair Authentication 

### ISSUE THREAD

**Issue by nyccto** [grouparoo/grouparoo#2532]
Snowflake supports key pair authentication
https://docs.snowflake.com/en/user-guide/key-pair-auth.html#configuring-key-pair-authentication
https://docs.snowflake.com/en/user-guide/nodejs-driver-use.html#label-nodejs-key-pair-authentication

Grouparoo should consider supporting this method of authentication since some organizations have strict security requirements that prohibit user/password based authentication.

**Comment by bleonard** [comment]
Thanks for the request, @nyccto! This is a great idea.
I think the best way to solve it is to allow Sources like Snowflake (and Destinations like Salesforce for that matter) to be able to use multiple Apps. Apps are a way to connect.
So we will look into creating a new `snowflake-keypair` app.
This also fits in well with the plan to enable OAuth on many services.

**Comment by bleonard** [comment]
Added to the Roadmap https://github.com/orgs/grouparoo/projects/26

**Comment by bleonard** [comment]
This is included in the `0.7.4` release!

**Comment by bleonard** [comment]
Example config
```
{
  "class": "App",
  "id": "warehouse",
  "name": "Warehouse",
  "type": "snowflake-keypair",
  "options": {
    "schema": "PUBLIC",
    "account": "xyz43169.us-east-1",
    "database": "ANALYTICS_DB",
    "private_key": "-----BEGIN PRIVATE KEY-----\nMIIE....7F88=\n-----END PRIVATE KEY-----\n",
    "username": "READ_ONLY_USER",
    "warehouse": "COMPUTE_WH"
  }
}
```
